### PR TITLE
Task/upgrade csi ver

### PR DESCRIPTION
**Pull request by bvered**

### CODE REVIEW

**Comment by bvered** [comment]
[Fix operator code to CSI 1.11](https://trello.com/c/HA6TaQr3/59-fix-operator-code-to-csi-111)

**Review by tyichye19 — APPROVED** [review]
LGTM

File: hack/build-operator-catalog.sh [hunk 1]
```diff
@@ -19,7 +19,15 @@
 set -e
 
 source hack/common.sh
-source hack/ensure-opm.sh
+#source hack/ensure-opm.sh
```
Comment (tyichye19) [inline hack/build-operator-catalog.sh:22]: why in comment? mistake or is there a problem with the opm and CSI?

File: hack/build-operator-bundle.sh [hunk 2]
```diff
@@ -26,3 +26,30 @@ docker build -f bundle.Dockerfile -t "${BUNDLE_FULL_IMAGE_NAME}" .
...
+      git clone "${DEFAULT_CSI_GIT_PATH}"
+      cd "${DEFAULT_CSI_LOCAL_PATH}"
+  else
+      cd "${DEFAULT_CSI_LOCAL_PATH}"
+      git pull
+  fi
+
+  cd "${DEFAULT_CSI_DOCKERFILE_PATH}"
+  echo "Building and pushing CSI bundle image using ${DEFAULT_CSI_DOCKERFILE_PATH}/${DEFAULT_CSI_DOCKERFILE_NAME}"
+  docker build -f "${DEFAULT_CSI_DOCKERFILE_NAME}" -t "${DEFAULT_CSI_DEVELOP_IMAGE}:${IMAGE_TAG}" .
```
Comment (tyichye19) [inline hack/build-operator-bundle.sh:47]: if I understand correctly, for non-GA versions we are downloading `develop` tag and building the bundle image using the dockerfile located in the relevant image version folder, right?
But the operator image in the CSV will be the GA one, which still does not exist no? do we need to edit the CSV file as we did for the CR, before building the bundle?

File: hack/ensure-blockcsi-cryaml.sh [hunk 3]
```diff
@@ -24,10 +24,17 @@ CSI_CR_PATH="config/manager/${BLOCK_CSI_CR_FILE}"
...
         curl -JL "${CSI_CR_URL}" -o "${CSI_CR_PATH}"
-        curl -JL "${CSI_CR_URL}" -o "${CSI_SAMPLES_CR_PATH}"
 else
-        echo "Using the cached CR file ${BLOCK_CSI_CR_FILE}"
+        echo "CSI tag doesn't exist yet, downloading develop version"
+        CSI_CR_URL="https://raw.githubusercontent.com/IBM/ibm-block-csi-operator/develop/config/samples/${BLOCK_CSI_CR_FILE}"
+        curl -JL "${CSI_CR_URL}" -o "${CSI_CR_PATH}"
+        sed -i 's/ibmcom\/ibm-block-csi-driver-controller/stg-artifactory.xiv.ibm.com:5030\/ibm-block-csi-driver-controller-amd64/g' "${CSI_CR_PATH}"
+        sed -i 's/ibmcom\/ibm-block-csi-driver-node/stg-artifactory.xiv.ibm.com:5030\/ibm-block-csi-driver-node-amd64/g' "${CSI_CR_PATH}"
+        sed -i 's/tag: "1.11.0"/tag: "latest"/g' "${CSI_CR_PATH}"
```
Comment (bvered) [inline hack/ensure-blockcsi-cryaml.sh:36]: replace 1.11 with version param

File: hack/ensure-blockcsi-cryaml.sh [hunk 4]
```diff
@@ -24,10 +24,17 @@ CSI_CR_PATH="config/manager/${BLOCK_CSI_CR_FILE}"
...
+        echo "Downloading the IBM Block CSI CR file on version ${BLOCK_CSI_RELEASE} ..."
         curl -JL "${CSI_CR_URL}" -o "${CSI_CR_PATH}"
-        curl -JL "${CSI_CR_URL}" -o "${CSI_SAMPLES_CR_PATH}"
 else
-        echo "Using the cached CR file ${BLOCK_CSI_CR_FILE}"
+        echo "CSI tag doesn't exist yet, downloading develop version"
+        CSI_CR_URL="https://raw.githubusercontent.com/IBM/ibm-block-csi-operator/develop/config/samples/${BLOCK_CSI_CR_FILE}"
+        curl -JL "${CSI_CR_URL}" -o "${CSI_CR_PATH}"
+        sed -i 's/ibmcom\/ibm-block-csi-driver-controller/stg-artifactory.xiv.ibm.com:5030\/ibm-block-csi-driver-controller-amd64/g' "${CSI_CR_PATH}"
+        sed -i 's/ibmcom\/ibm-block-csi-driver-node/stg-artifactory.xiv.ibm.com:5030\/ibm-block-csi-driver-node-amd64/g' "${CSI_CR_PATH}"
```
Comment (bvered) [inline hack/ensure-blockcsi-cryaml.sh:35]: replace with DEFAULT_CSI_DEVELOP_REGISTRY="stg-artifactory.xiv.ibm.com:5030"

File: hack/bundle-manifests.sh [hunk 5]
```diff
@@ -35,7 +35,7 @@ dependencies:
   - type: olm.package
     value:
       packageName: ibm-block-csi-operator
-      version: ">=${BLOCK_CSI_RELEASE:1}"
+      version: ">=${CSI_RELEASE:1}"
```
Comment (tyichye19) [inline hack/bundle-manifests.sh:38]: I saw you create a const `CSI_RELEASE_NUMBER` with the same code - can you use it here?

**Review by tyichye19 — APPROVED** [review]
LGTM

**Comment by bvered** [comment]
[Add develop CSI images to ODF bundle (for tests)](https://trello.com/c/tqEQc6eA/189-add-develop-csi-images-to-odf-bundle-for-tests)

**Comment by bvered** [comment]
[add csi develop operator latest image to catalog](https://trello.com/c/g0ds2HU3/194-add-csi-develop-operator-latest-image-to-catalog)

**Pull request merged**